src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py

```diff
@@ -467,3 +468,405 @@ def _test_flexible_server_replica_mgmt(self, database_engine, resource_group):
+                             self.get_subscription_id(), resource_group, vnet_name, 'Subnet' + servers[0][6:]))
+
+        self.assertEqual(show_result_2['delegatedSubnetArguments']['subnetArmResourceId'],
+                         '/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Network/virtualNetworks/{}/subnets/{}'.format(
+                             self.get_subscription_id(), resource_group, vnet_name_2, 'Subnet' + servers[1][6:]))
+
+        # delete all servers
+        self.cmd('{} flexible-server delete -g {} -n {} --yes'.format(database_engine, resource_group, servers[0]),
+                 checks=NoneCheck())
+
+        self.cmd('{} flexible-server delete -g {} -n {} --yes'.format(database_engine, resource_group, servers[1]),
+                 checks=NoneCheck())
+
+        time.sleep(15 * 60)
+
+        # remove delegations from all vnets
+        self.cmd('network vnet subnet update -g {} --name {} --vnet-name {} --remove delegations'.format(resource_group,
+                                                                                                         'Subnet' + servers[0][6:],
+                                                                                                         vnet_name))
+
+        self.cmd('network vnet subnet update -g {} --name {} --vnet-name {} --remove delegations'.format(resource_group,
+                                                                                                         'Subnet' + servers[1][6:],
+                                                                                                         vnet_name_2))
+
+        # remove all vnets
+        self.cmd('network vnet delete -g {} -n {}'.format(resource_group, vnet_name))
+        self.cmd('network vnet delete -g {} -n {}'.format(resource_group, vnet_name_2))
+
+    def _test_flexible_server_vnet_mgmt_supplied_vname_and_subnetname(self, database_engine, resource_group):
+
+        # flexible-server create
+        if self.cli_ctx.local_context.is_on:
+            self.cmd('local-context off')
+
+        if database_engine == 'postgres':
+            location = self.postgres_location
+        elif database_engine == 'mysql':
+            location = self.mysql_location
+
+        vnet_name = 'clitestvnet5'
+        subnet_name = 'clitestsubnet5'
+        address_prefix = '10.0.0.0/16'
+        subnet_prefix_1 = '10.0.0.0/24'
+        vnet_name_2 = 'clitestvnet6'
+
+        # flexible-servers
+        servers = ['testvnetserver5' + database_engine, 'testvnetserver6' + database_engine]
+
+        # Case 1 : Provision a server with supplied Vname and subnet name that exists.
+
+        # create vnet and subnet. When vnet name is supplied, the subnet created will be given the default name.
+        vnet_result = self.cmd('network vnet create -n {} -g {} -l {} --address-prefix {} --subnet-name {} --subnet-prefix {}'
```

Comment:
could you use VirtualNetworkPreparer() as shown in https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py#L468 for vnet?

It will not record network info in your record file and remove the dependency with network in CI check.

Reply:
Removed it for the tests wherever applicable. However, did not remove it in some of the tests where we have certain restrictions around how the vnet is created, named and in which rg it is provisioned.